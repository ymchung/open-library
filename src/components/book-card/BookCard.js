import React from 'react';
import { string, number, func } from 'prop-types';

const namespace = 'book-card';

const BookCard = ({ book, title, authorName, year, handleOnBook }) => (
    <div className={`${namespace} card`}>
        <div className="card-body">
            <a href="/#" className={`${namespace}__link`} onClick={e => handleOnBook(e, book)} data-testid={`${namespace}__link`}>
                <h3 className={`${namespace}__title`}>{title}</h3>
                <p className={`${namespace}__subtitle`}>de {authorName}</p>
                <p className={`${namespace}__text`}>Publicado por primera vez en {year}</p>
            </a>
        </div>
    </div>
);

BookCard.propTypes = {
    book: string,
    title: string,
    authorName: string,
    year: number,
    handleOnBook: func.isRequired,
};

export default BookCard;