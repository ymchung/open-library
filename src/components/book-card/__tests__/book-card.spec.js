import { render, fireEvent } from '@testing-library/react';
import BookCard from '../BookCard';

describe('BookCard Component', () => {
    const props = {
        book: 'AB123',
        title: 'Harry Potter',
        authorName: 'J. K. Rowling',
        year: 1999,
        handleOnBook: jest.fn(),
    }

    test('Component should render correctly', () => {
        const { container } = render(<BookCard {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    test('On click component should call handleOnBook function', () => {
        const { getByTestId } = render(<BookCard {...props} />);
        fireEvent.click(getByTestId('book-card__link'));

        expect(props.handleOnBook).toHaveBeenCalledTimes(1);
    });
});