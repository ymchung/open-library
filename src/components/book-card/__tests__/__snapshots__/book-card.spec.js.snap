// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard Component Component should render correctly 1`] = `
<div
  class="book-card card"
>
  <div
    class="card-body"
  >
    <a
      class="book-card__link"
      data-testid="book-card__link"
      href="/#"
    >
      <h3
        class="book-card__title"
      >
        Harry Potter
      </h3>
      <p
        class="book-card__subtitle"
      >
        de 
        J. K. Rowling
      </p>
      <p
        class="book-card__text"
      >
        Publicado por primera vez en 
        1999
      </p>
    </a>
  </div>
</div>
`;
